### PR TITLE
Decrypting payload

### PR DESCRIPTION
**Pull request by kaipakartik**

### CODE REVIEW

File: src/runtime/pay-flow-test.js [hunk 1]
```diff
@@ -30,6 +30,8 @@ import {
 import {PurchaseData, SubscribeResponse} from '../api/subscribe-response';
 import {UserData} from '../api/user-data';
 import * as sinon from 'sinon';
+import {Xhr} from '../utils/xhr';
```
Comment (dvoytenko) [inline src/runtime/pay-flow-test.js]: Bring one line up.

File: src/runtime/pay-flow-test.js [hunk 2]
```diff
@@ -282,6 +287,49 @@ describes.realWin('PayCompleteFlow', {}, env => {
...
+    it('should start flow on correct payment response as encrypted obj', () => {
+      const encryptedResponse = {
+        redirectEncryptedCallbackData: INTEGR_DATA_STRING,
+        environment: 'PRODUCTION',
+      };
+      const result = new ActivityResult(ActivityResultCode.OK,
+          encryptedResponse,
+          'POPUP', location.origin, true, true);
+      const xhrFetchStub = sandbox.stub(Xhr.prototype, 'fetch',
+        () => Promise.resolve({json:() => Promise.resolve(INTEGR_DATA_OBJ_DECODED)}))
```
Comment (dvoytenko) [inline src/runtime/pay-flow-test.js]: looks like semicolon missing.

File: src/runtime/pay-flow-test.js [hunk 3]
```diff
@@ -282,6 +287,49 @@ describes.realWin('PayCompleteFlow', {}, env => {
...
+        expect(JSON.parse(response.raw)).to.deep
+            .equal(JSON.parse(atob(INTEGR_DATA_STRING))['swgCallbackData']);
+        expect(completeStub).to.not.be.called;
+        response.complete();
+        expect(completeStub).to.be.calledOnce;
+
+        // Verify xhr call
+        expect(xhrFetchStub).to.be.calledOnce;
+        expect(xhrFetchStub).to.be.calledWith(
+          'https://pay.google.com/gp/p/apis/buyflow/process', ({
```
Comment (dvoytenko) [inline src/runtime/pay-flow-test.js]: Can you also add a test for a non-prod URL?

File: src/runtime/pay-flow-test.js [hunk 4]
```diff
@@ -282,6 +287,49 @@ describes.realWin('PayCompleteFlow', {}, env => {
         expect(completeStub).to.be.calledOnce;
       });
     });
+
+    it('should start flow on correct payment response as encrypted obj', () => {
+      const encryptedResponse = {
+        redirectEncryptedCallbackData: INTEGR_DATA_STRING,
```
Comment (dvoytenko) [inline src/runtime/pay-flow-test.js]: Just use a fake string here. E.g. `"<ENCRYPTED>"`.

File: src/runtime/pay-flow-test.js [hunk 5]
```diff
@@ -282,6 +287,49 @@ describes.realWin('PayCompleteFlow', {}, env => {
...
+        expect(completeStub).to.be.calledOnce;
+
+        // Verify xhr call
+        expect(xhrFetchStub).to.be.calledOnce;
+        expect(xhrFetchStub).to.be.calledWith(
+          'https://pay.google.com/gp/p/apis/buyflow/process', ({
+            method: 'post',
+            headers: {'Accept': 'text/plain, application/json'},
+            credentials: 'include',
+            body: INTEGR_DATA_STRING,
```
Comment (dvoytenko) [inline src/runtime/pay-flow-test.js]: Ditto:  `"<ENCRYPTED>"`.

File: src/runtime/pay-flow.js [hunk 6]
```diff
@@ -24,6 +24,7 @@ import {UserData} from '../api/user-data';
 import {acceptPortResult} from '../utils/activity-utils';
 import {parseJson} from '../utils/json';
 import {feArgs, feOrigin, feUrl} from './services';
+import {Xhr} from '../utils/xhr';
```
Comment (dvoytenko) [inline src/runtime/pay-flow.js]: Sort.

File: src/runtime/pay-flow.js [hunk 7]
```diff
@@ -175,19 +177,35 @@ export class PayCompleteFlow {
 
 
 /**
+  *@param @const {!Window} win
```
Comment (dvoytenko) [inline src/runtime/pay-flow.js]: formatting... and no `@const` here.

File: src/runtime/pay-flow.js [hunk 8]
```diff
@@ -175,19 +177,35 @@ export class PayCompleteFlow {
...
  * @param {!web-activities/activity-ports.ActivityPort} port
  * @param {function():!Promise} completeHandler
+
```
Comment (dvoytenko) [inline src/runtime/pay-flow.js]: remove this line

File: src/runtime/pay-flow.js [hunk 9]
```diff
@@ -231,6 +249,19 @@ export function parseSubscriptionResponse(data, completeHandler) {
...
+   * Returns the decryption url to be used to decrypt the encrypted payload.
+   *
+   * @param {!string} environment
+   * @return {!string} The decryption url
+   */
+function getDecryptionUrl(environment) {
+  if (environment == 'PRODUCTION') {
+    return 'https://pay.google.com/gp/p/apis/buyflow/process';
+  }
+    return 'https://pay.sandbox.google.com/gp/p/apis/buyflow/process';
```
Comment (dvoytenko) [inline src/runtime/pay-flow.js]: indent

File: src/runtime/pay-flow-test.js (hunk 3 above)
Comment (kaipakartik) [reply src/runtime/pay-flow-test.js]: Done

File: src/runtime/pay-flow-test.js (hunk 4 above)
Comment (kaipakartik) [reply src/runtime/pay-flow-test.js]: Done

File: src/runtime/pay-flow-test.js (hunk 5 above)
Comment (kaipakartik) [reply src/runtime/pay-flow-test.js]: Done

File: src/runtime/pay-flow.js (hunk 6 above)
Comment (kaipakartik) [reply src/runtime/pay-flow.js]: Done

File: src/runtime/pay-flow.js (hunk 7 above)
Comment (kaipakartik) [reply src/runtime/pay-flow.js]: Done

**Comment by kaipakartik** [comment]
Fixed comments.

**Pull request merged**